src/core/lib/surface/server.cc

```diff
@@ -568,8 +579,9 @@ static void publish_new_rpc(void* arg, grpc_error* error) {
   }
 
   gpr_atm_no_barrier_store(&calld->state, PENDING);
-  if (rm->pending_head == nullptr) {
-    rm->pending_tail = rm->pending_head = calld;
+  if (rm->pending_head.load(std::memory_order_relaxed) == nullptr) {
+    rm->pending_head.store(calld, std::memory_order_release);
```

Comment:
You should add a comment explaining what this release matches with (ditto on the other end)

Reply:
Thank you David! Sure, added notes pair-wise in both locations explaining the ordering.

Reply:
But now see my post below -- I think this pair is unnecessary.